src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinString = '';
+    let stringSeparator = '';
+
+    separator === undefined
+      ? stringSeparator = ','
+      : stringSeparator = '' + separator;
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = '';
+      }
```

Comment:
let's add element to the result string only if it's not null or undefined instead of setting empty string to the element